generator.py

```diff
@@ -95,7 +95,8 @@
 
 protocol_defs = load_services(protocol_defs_path)
 protocol_defs = sorted(protocol_defs, key=lambda proto_def: proto_def['id'])
-if not validate_services(protocol_defs, schema_path):
+check_service_id = proto_path_arg is None
```

Comment:
I prefer not using an argument which is designed for another purpose to implicitly imply "DO not check id". Therefore, I suggest that you put an explicit boolean option to turn it off.